File: Core/Models/Player.cs
using Core.Enums;
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Models
{
    public sealed class Player : _ParentPerson
    {
        public PositionEnum Position { get; set; }
        public bool HasInjury { get; set; }
        public int PlayedMinutes { get; set; }
        public int YellowCards { get; set; }
        public int RedCards { get; set; }
        public int Goals { get; set; }
        public int Assists { get; set; }
        public int PlayedMatches { get; set; }
        public int PlayerNumber { get; set; }
        [ForeignKey("PlayerClub")]
        public Guid PlayerClubId { get; set; }
        public SportsClub? PlayerClub { get; set; }
        private Player() : base() { }
        public Player(
            PositionEnum position,DateTime contractFrom, DateTime contractTo, string firstName, string lastName, 
            bool hasInjury = false, int playedMinutes = 0 ,int yellowCard = 0, int redCard = 0, int goals = 0, int assit = 0, int playedMatches = 0
            , int playerNumber = -1):base(contractFrom, contractTo, firstName, lastName)
        {
            Position = position;
            HasInjury = hasInjury;
            PlayedMinutes = playedMinutes;
            YellowCards = yellowCard;
            RedCards = redCard;
            Goals = goals;
            Assists = assit;
            PlayedMatches = playedMatches;
            PlayerNumber = playerNumber;
        }
        public void updateByModel(Player player)
        {
            this.ContractFrom = player.ContractFrom;
            this.ContractTo = player.ContractTo;
            this.FirstName = player.FirstName;
            this.LastName = player.LastName;
            this.Position = player.Position;
            this.HasInjury = player.HasInjury;
            this.PlayedMinutes = player.PlayedMinutes;
            this.YellowCards = player.YellowCards;
            this.RedCards = player.RedCards;
            this.Goals = player.Goals;
            this.Assists = player.Assists;
            this.PlayedMatches = player.PlayedMatches;
            this.PlayerNumber = player.PlayerNumber;
            this.UpdateAt = DateTime.Now;
        }
    }
}

File: FrontEnd_Angular/src/app/shared/Interfaces/Player.ts
export interface Player {
  Id: string;
  Status: number;
  CreateAt: Date;
  UpdateAt: Date;
  ContractFrom: Date;
  ContractTo: Date;
  FirstName: string;
  LastName: string;

  position: number;
  hasInjury: boolean;
  PlayedMinutes: number;
  YellowCards: number;
  RedCards: number;
  Goals: number;
  Assists: number;
  PlayedMatches: number;
  PlayerClubId: string;
  PlayerNumber: number;
}
export interface PlayerDetailsDTO {
  id: string;
  firstName: string;
  lastName: string;
  contractFrom: Date;
  contractTo: Date;
  clubName: string;
  hasInjury: boolean;
  playedMinutes: number;
  yellowCards: number;
  redCards: number;
  goals: number;
  assists: number;
  playedMatches: number;
  position: number;
  playerNumber: number;
}
export interface EditPlayerCommand {
  playerId: string;
  contractFrom: Date;
  contractTo: Date;
  firstName: string;
  lastName: string;

  position: number;
  hasInjury: boolean;
  playedMinutes: number;
  yellowCards: number;
  redCards: number;
  goals: number;
  assists: number;
  playedMatches: number;
  playerNumber: number;
}
export interface CreatePlayerCommand {
  clubId: string;
  contractFrom: Date;
  contractTo: Date;
  firstName: string;
  lastName: string;

  position: number;
  hasInjury: boolean;
  playedMinutes: number;
  yellowCards: number;
  redCards: number;
  goals: number;
  assists: number;
  playedMatches: number;
  playerNumber: number;
}
export function GetPlayerPosition(position: number): string {
  switch (position) {
    case 0:
      return 'Bramkarz';
    case 1:
      return 'Środkowy obrońca';
    case 2:
      return 'Prawy obrońca';
    case 3:
      return 'Lewy obrońca';
    case 4:
      return 'Środkowy pomocnik';
    case 5:
      return 'Prawy pomocnik';
    case 6:
      return 'Lewy pomocnik';
    case 7:
      return 'Środkowy napastnik';
    case 8:
      return 'Prawy napastnik';
    case 9:
      return 'Lewy napastnik';
    default:
      return 'Nieustalona';
  }
}
export function mapPlayerDetailsToEditPlayerCommand(
  dto: PlayerDetailsDTO,
): EditPlayerCommand {
  return {
    playerId: dto.id,
    contractFrom: dto.contractFrom,
    contractTo: dto.contractTo,
    firstName: dto.firstName,
    lastName: dto.lastName,
    position: dto.position,
    hasInjury: dto.hasInjury,
    playedMinutes: dto.playedMinutes,
    yellowCards: dto.yellowCards,
    redCards: dto.redCards,
    goals: dto.goals,
    assists: dto.assists,
    playedMatches: dto.playedMatches,
    playerNumber: dto.playerNumber,
  };
}
export function mapEditPlayerCommandToPlayerDetails(
  dto: EditPlayerCommand,
): PlayerDetailsDTO {
  return {
    id: dto.playerId,
    clubName: '',
    contractFrom: dto.contractFrom,
    contractTo: dto.contractTo,
    firstName: dto.firstName,
    lastName: dto.lastName,
    position: dto.position,
    hasInjury: dto.hasInjury,
    playedMinutes: dto.playedMinutes,
    yellowCards: dto.yellowCards,
    redCards: dto.redCards,
    goals: dto.goals,
    assists: dto.assists,
    playedMatches: dto.playedMatches,
    playerNumber: dto.playerNumber,
  };
}
